gateway-ha/src/main/java/io/trino/gateway/ha/handler/QueryIdCachingProxyHandler.java

```diff
@@ -252,13 +285,26 @@ private boolean isPathWhiteListed(String path)
                 || extraWhitelistPaths.stream().anyMatch(s -> path.startsWith(s));
     }
 
-    public boolean handleAuthRequest(HttpServletRequest request)
+    @Override
+    public Optional<Cookie> deleteCookieIfNeeded(HttpServletRequest clientRequest)
     {
-        return true;
+        if (!removeCookiePaths.contains(clientRequest.getRequestURI())) {
+            // TODO: support custom rules for determining if a SessionId cookie should be removed and provide current
```

Comment:
??

Reply:
In which cases would you need custom rules?

Reply:
this note is here b/c I was thinking about supporting things like the Trino memory connector. For example
```
CREATE TABLE memory.default.cached_result AS SELECT ...
```
The admin in this case might want to add a cookie to make sure every subsequent query in this session goes to the cluster where this user has created their memory connector tables, and remove the cookie once they have all been `DROP`'d. 

I'll remove this TODO to an issue, after all we may choose a different approach if we find a need for this functionality some day.